tests/car_tests.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+pub mod common;
+
+use std::path::Path;
+
+use anyhow::*;
+use cid::{
+    multihash::{self, MultihashDigest},
+    Cid,
+};
+use futures::StreamExt;
+use fvm_ipld_car::{CarHeader, CarReader};
+use fvm_ipld_encoding::DAG_CBOR;
+use rand::{rngs::SmallRng, Rng, SeedableRng};
+use tempfile::NamedTempFile;
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+use crate::common::cli;
+
+#[tokio::test]
+async fn forest_cli_car_concat() -> Result<()> {
```

Comment:
I don't think these tests are useful. Correct me if I'm wrong, but they just check if the output file is a valid CAR file, right? I can imagine lots of incorrect implementations that would pass these tests.

Let's either remove the tests or make them more rigorous.

Reply:
Unlike unit tests, the main purpose of these bin tests ensure the executable runs without failures, and the assertions ensure the output file exists and can be parsed by `CarReader`.  Any suggestions on adding more checks of the output `car` file?

Reply:
Would love unit tests that didn't use any files at all.

This PR is about a function that computes the union of a list of sets. We can test that thoroughly with quickcheck. Generate arbitrary sets of data and verify that the union contains all the elements of the input sets with no duplicates.

∀AB. A⊆(A∪B)
∀AB. B⊆(A∪B)
∀AB. A∪B = B∪A
∀ABC. (A ∪ B) ∪ C = A ∪ (B ∪ C)
∀A. A∪A = A

Reply:
asides: 
- I think we should invest in some mock IPLD/ car creation utils rather than checking in test files
- I think we should have better guidelines on when and where to write integration tests. Part of the "engineering values" piece